### PR TITLE
Fix component startup time.

### PR DESCRIPTION
**Pull request by ryanwilson**
The `configureWithApp:` call shouldn't have been added - the
component registration system can provide proper configuration
instead of relying on a static `configureWithApp:` call. Many of
the SDKs relied on singletons being available there, which wasn't
necessary. Instead, this should happen in the component creation
block and use the instantiationTiming parameter on
registerInternalLibrary.

This replaces #3147 since other Core and IID changes were made since then.

### CODE REVIEW

File: Firebase/Auth/Source/Auth/FIRAuth.m [hunk 1]
```diff
@@ -1884,19 +1884,12 @@ - (BOOL)getUser:(FIRUser *_Nullable *)outUser
     return [[FIRAuth alloc] initWithApp:container.app];
   };
   FIRComponent *authInterop = [FIRComponent componentWithProtocol:@protocol(FIRAuthInterop)
+                                              instantiationTiming:FIRInstantiationTimingAlwaysEager
```
Comment (paulb777) [inline Firebase/Auth/Source/Auth/FIRAuth.m]: Why AlwaysEager?

File: Firebase/Messaging/FIRMessaging.m [hunk 2]
```diff
@@ -233,28 +240,19 @@ + (void)load {
                                  withInstanceID:[FIRInstanceID instanceID]
                                withUserDefaults:[GULUserDefaults standardUserDefaults]];
     [messaging start];
+    [messaging configureNotificationSwizzlingIfEnabled];
     return messaging;
   };
   FIRComponent *messagingProvider =
       [FIRComponent componentWithProtocol:@protocol(FIRMessagingInstanceProvider)
-                      instantiationTiming:FIRInstantiationTimingLazy
+                      instantiationTiming:FIRInstantiationTimingEagerInDefaultApp
```
Comment (paulb777) [inline Firebase/Messaging/FIRMessaging.m]: Why this change from lazy to eager?

File: Firebase/Auth/Source/Auth/FIRAuth.m (hunk 1 above)
Comment (ryanwilson) [reply Firebase/Auth/Source/Auth/FIRAuth.m]: It mimics the previous behaviour removed on the previous line 1897:

 https://github.com/firebase/firebase-ios-sdk/pull/4137/files#diff-f33c561b2a3430b947e6da076758c6d9L1897

every time `[FIRApp configure]` is called, we called `[FIRAuth authWithApp:]` which instantiated the instance. 

The reason it's `AlwaysEager` and not `EagerInDefaultApp` is because `FIRAuth` works with multiple `FIRApp`s.

File: Firebase/Messaging/FIRMessaging.m (hunk 2 above)
Comment (ryanwilson) [reply Firebase/Messaging/FIRMessaging.m]: It mimics the previous behaviour of `configureWithApp:`, where Messaging would instantiate itself as soon as `configureWithApp:` is called.

File: Firebase/Messaging/FIRMessaging.m [hunk 3]
```diff
@@ -233,28 +240,19 @@ + (void)load {
...
                              dependencies:@[ analyticsDep ]
                            creationBlock:creationBlock];
 
   return @[ messagingProvider ];
 }
 
-+ (void)configureWithApp:(FIRApp *)app {
```
Comment (chliangGoogle) [inline Firebase/Messaging/FIRMessaging.m]: do we need to fix the same for other SDK as well, such as InstanceID, remote config etc?

File: Firebase/Auth/Source/Auth/FIRAuth.m (hunk 1 above)
Comment (renkelvin) [reply Firebase/Auth/Source/Auth/FIRAuth.m]: Yeah, `FIRAuth` works with multiple `FIRApp` s.

File: Firebase/Messaging/FIRMessaging.m (hunk 3 above)
Comment (ryanwilson) [reply Firebase/Messaging/FIRMessaging.m]: Apologies for the delay - IID was fixed in #4030, will be addressing the other SDKs in a separate PR.

**Comment by ryanwilson** [comment]
Merged laster to re-run travis and merge this shortly.

**Comment by ryanwilson** [comment]
Merging this now since everything is green - will fix anything if there was a conflict since last merge.
